services/aster-nix/src/net/iface/util.rs

```diff
@@ -46,30 +48,39 @@ impl BindPortConfig {
 }
 
 pub fn spawn_background_poll_thread(iface: Arc<dyn Iface>) {
-    // FIXME: use timer or wait_timeout when timer is enable.
     let task_fn = move || {
-        debug!("spawn background poll thread");
+        trace!("spawn background poll thread for {}", iface.name());
+        let wait_queue = iface.polling_wait_queue();
         loop {
-            let next_poll_time = if let Some(next_poll_time) = iface.next_poll_at_ms() {
-                next_poll_time
+            let next_poll_at_ms = if let Some(next_poll_at_ms) = iface.next_poll_at_ms() {
+                next_poll_at_ms
+            }
+            // Sleep for a long time, if `next_poll_at_ms` is `None`.
+            else if let Some(next_poll_at_ms) = wait_queue
+                .wait_until_or_timeout(|| iface.next_poll_at_ms(), &Duration::from_secs(30))
+            {
+                next_poll_at_ms
             } else {
-                Thread::yield_now();
                 continue;
             };
-            let now = read_monotonic_milli_seconds();
-            if now > next_poll_time {
-                // FIXME: now is later than next poll time. This may cause problem.
+
+            let now_as_ms = read_monotonic_milli_seconds();
+
+            // FIXME: Ideally, we should perform the `poll` just before `next_poll_at_ms`. However, this
+            // approach may result in a spinning busy loop if the `poll` operation yields no results. To
+            // mitigate this issue, we have opted to assign a high priority to the polling thread, ensuring
+            // that the `poll` runs as soon as possible.
+            // For a more in-depth discussion, please refer to the following link: https://github.com/asterinas/asterinas/pull/630.
```

Comment:
```suggestion
            // For a more in-depth discussion, please refer to the following link:
            // https://github.com/asterinas/asterinas/pull/630#discussion_r1496817030
```
Maybe this looks a bit better?